### ISSUE TITLE
Every invocation of terraform plan/apply updates the artifactory_proxy resource even when no changes

### ISSUE THREAD

**Issue by mymasse** [jfrog/terraform-provider-artifactory#941]
**Describe the bug**

We define the following configuration:
```hcl
resource "artifactory_proxy" "this" {
  key              = "my-proxy"
  host             = "my-proxy.com"
  port             = 3128
  platform_default = false
}
```

Every plan/apply now lists the following:

```
# artifactory_proxy.this will be updated in-place
~ resource "artifactory_proxy" "this" {
  # (7 unchanged attributes hidden)
}
```

Artifactory Version: 7.77.6 rev 77706900
Terraform Version: 1.8.0
Artifactory Provider: 10.3.1
Platform Provider: 1.4.1

**Requirements for and issue**
- [x] A description of the bug
- [x] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue). **If this is not supplied, this issue will likely be closed without any effort expended.**
- [x] Your version of artifactory (you can `curl` it at `$host/artifactory/api/system/version`
- [x] Your version of terraform
- [x] Your version of terraform provider

**Expected behavior**
If nothing changes in the configuration the ressource should not be updated

**Additional context**
N/A

**Comment by alexhung** [comment]
@mymasse Are you upgrading from a previous version of the provider? Or is this happening with new resource configuration for proxy?

**Comment by mymasse** [comment]
The proxy as been defined for a while and we have been keeping the provider up to date, but to my recollection it's always complained about beeing updated during all subsequent plan/apply.

We have not tried on a new resource

**Comment by alexhung** [comment]
@mymasse Thanks! I'm investigating now.

**Comment by mymasse** [comment]
Confirmed bug is fixed with provider 10.7.2
